omegaconf/base.py

```diff
@@ -4,10 +4,33 @@
 from enum import Enum
 from typing import Any, Dict, Iterator, Optional, Tuple, Type, Union
 
-from ._utils import ValueKind, _get_value, format_and_raise, get_value_kind
+from ._utils import (
+    ValueKind,
+    _get_value,
+    format_and_raise,
+    get_value_kind,
+    update_string,
+)
 from .errors import ConfigKeyError, MissingMandatoryValue, UnsupportedInterpolationType
 
 
+@dataclass
+class InterpolationRange:
+    """Used to store start/stop indices of interpolations being evaluated"""
+
+    start: int = -1
+    stop: int = -1
+
+
+class ParseError(Exception):
+    """Thrown when unable to parse a complex interpolation"""
+
+    @staticmethod
+    def assert_(condition: Any) -> None:
+        if not condition:
+            raise ParseError()
```

Comment:
I don't think this logic belongs in the exception class.

Reply:
Changed in fb08218